grails-app/domain/org/pih/warehouse/order/OrderAdjustment.groovy

```diff
@@ -100,27 +99,28 @@ class OrderAdjustment implements Serializable, Comparable<OrderAdjustment> {
         }
     }
 
-    Boolean getIsInvoiced() {
-        return !postedPurchaseInvoiceItems.empty
+    Boolean getIsFullyInvoiced() {
+        return Math.abs(unitPriceOnPostedInvoices) >= Math.abs(totalAdjustments)
     }
 
     OrderAdjustmentInvoiceStatus getDerivedPaymentStatus() {
-        Boolean fullyInvoiced = invoicedUnitPrice == totalAdjustments
-
-        if (!isInvoiced) {
+        if (postedPurchaseInvoiceItems.empty) {
```

Comment:
Could you add some comments inside this method (especially on the latter ifs checks)? Looking at the ordering of the if checks it overall makes sense and should work now, but I am afraid, that if someone starts adjusting here something, then it might easily get broken.